core/src/lib.rs

```diff
@@ -1,4 +1,4 @@
-#![feature(associated_type_bounds)]
+#![feature(associated_type_bounds, type_name_of_val)]
```

Comment:
Why needed?

Reply:
That was missed when I was cleaning up from the debugging work, removed now.